### ISSUE TITLE
generate yaml files from spreadsheet

### ISSUE THREAD

**Issue by sannatitus** [SainsburyWellcomeCentre/WAZP#85]
When I used the "generate yaml files from spreadsheet" for the first time, it automatically created .yaml files for most of the entries in the spreadsheet (I didn't have to click boxes or anything). This has caused leaf cutter ant metadata files to appear in my jewel wasp folders (but not cricket nor NMR metadata?) :( 

Not sure if this is possible, but would there be a way to match the videos in a specific folder (ie, the one in input_config) to the values within the master-list.xlsx 'file' column, to generate .yaml files automatically? For example, I want the files in X:\processed\LondonZoo\jewel-wasp_Ampulex-compressa\DLC_femaleandmale\jwasp_femaleandmale-Sanna-2022-09-12\videos\ to have their associated metadata files, but no other metadata files from the master-list present from videos in other folders. At the moment, it appears that one has to click one by one for which videos to convert but they just all went automatically. If it were one by one, it would still be easy to miss a video here and there. This would theoretically be solved by the check for missing metadata files button, but I'm too scared to click it and delete all the extra .yamls again :o 

Can we do this automatically by matching the file names of the videos that are present in the folder?

**Comment by sfmig** [comment]
Hi @sannatitus ,

Just a bit more context for these fixes:

I now made some changes so that we only generate YAML files for only rows in the spreadsheet for which there is a video in the video directory. 

I also added a tooltip for the 'check missing files button', to clarify that this button only adds rows to the table, but does not export them as YAML files (to do this, the user needs to click the export selected rows button).

Let me know if you were thinking of something else / something more! As always, feel free to add issues for any request (big or small)

**Comment by sannatitus** [comment]
@sfmig just for some clarification -
the 'check for missing metadata files' should theoretically add rows to the table within the wazp-dashboard for any videos that have missing metadata. would it be possible to check them with the checkbox, so they stand out? Otherwise, it's hard to tell if anything was added without scrolling to the bottom first and making a note of the last one.
Also, I'm not sure on the difference between the 'generate yaml files from spreadsheet' and the 'export selected rows as yaml'...

**Comment by sannatitus** [comment]
Hey all, just a note to say that the metadata export is extremely flawless!! I now have .yamls for every video within the ceph, all thanks to you guys! 

There are a couple, very small, details that I noticed throughout this process - which have been upated in my last comment. These 'issues' don't bother me, but I figured it was worth mentioning them just in case your team has thoughts of making this dashboard available to the public. 

Thank you all so, so much for your hardwork! Now onto the ROIs...
